File: 2171-removing-minimum-number-of-magic-beans/2171-removing-minimum-number-of-magic-beans.js
/**
 * @param {number[]} beans
 * @return {number}
 */
var minimumRemoval = function(beans) {
        let totalBeans = 0;
        for(let b of beans) totalBeans+=b;
        beans.sort(function(a,b){return a-b;})
        let minRemoved = Number.MAX_VALUE;
        for(let i=0;i<beans.length;i++){
            minRemoved = Math.min(minRemoved,totalBeans-(beans.length-i)*beans[i]);
        }
        return minRemoved;
};

File: 2171-removing-minimum-number-of-magic-beans/2171-removing-minimum-number-of-magic-beans.py
class Solution:
    def minimumRemoval(self, beans: List[int]) -> int:
        totalBeans = sum(beans)
        beans.sort()
        minRemoved = float('inf')
        for i,b in enumerate(beans):
            minRemoved = min(minRemoved,totalBeans-(len(beans)-i)*b)
        return minRemoved